apps/tangle-dapp/data/DelegationsPayouts/usePayouts.ts

```diff
@@ -0,0 +1,288 @@
+import { WebbError, WebbErrorCodes } from '@webb-tools/dapp-types/WebbError';
+import { useEffect, useState } from 'react';
+import { Subscription } from 'rxjs';
+
+import {
+  formatTokenBalance,
+  getPolkadotApiPromise,
+  getPolkadotApiRx,
+  getValidatorCommission,
+  getValidatorIdentity,
+} from '../../constants';
+import useFormatReturnType from '../../hooks/useFormatReturnType';
+import { Payout } from '../../types';
+
+export default function usePayouts(
+  address: string,
+  defaultValue: { payouts: Payout[] } = {
+    payouts: [],
+  }
+) {
+  const [payouts, setPayouts] = useState(defaultValue.payouts);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    let isMounted = true;
+    let sub: Subscription | null = null;
+
+    const subscribeData = async () => {
+      if (!address) {
+        if (isMounted) {
+          setPayouts([]);
+          setIsLoading(false);
+        }
+        return;
+      }
+
+      try {
+        const apiSub = await getPolkadotApiRx();
+        const apiPromise = await getPolkadotApiPromise();
+
+        if (!apiSub || !apiPromise) {
+          throw WebbError.from(WebbErrorCodes.ApiNotReady);
+        }
+
+        setIsLoading(true);
+
+        const nominations = await apiPromise.query.staking.nominators(address);
+        const myNominations = nominations.unwrap().targets;
+
+        sub = apiSub.query.staking.erasRewardPoints
+          .entries()
+          .subscribe(async (points) => {
+            const allRewards: {
+              era: number;
+              totalRewardPoints: number;
+              validator: string;
+              validatorRewardPoints: number;
+            }[] = [];
+
+            let validatorPayoutsPromises: Promise<Payout>[] = [];
+
+            if (myNominations.length > 0) {
+              myNominations.forEach((validator) => {
+                points.forEach((point) => {
+                  const era = Number(
+                    point[0].toHuman()?.toString().replace(/,/g, '')
+                  );
+
+                  if (era) {
+                    const rewards = point[1].toHuman();
+
+                    let validatorRewardPoints = 0;
+
+                    if (rewards) {
+                      const totalRewardPoints = parseFloat(
+                        rewards.total?.toString().replace(/,/g, '') ?? '0'
+                      );
+
+                      if (
+                        typeof rewards.individual === 'object' &&
+                        rewards.individual !== null
+                      ) {
+                        Object.entries(rewards.individual).forEach(
+                          ([key, value]) => {
+                            if (key === validator.toString()) {
+                              validatorRewardPoints = Number(value);
+                            }
+                          }
+                        );
```

Comment:
Please correct me if I am understanding wrong, I'm assuming that we're looking for a single specific entry here, and if/once we find it, we set the `validatorRewardPoints` to its value.

In that context, could convert this to a `for..of` loop, and then `break` when you find the `key === validator.toString()`, that way the iteration doesn't continue when not needed:

```ts
for (const [key, value] of Object.entries(rewards.individual)) {
  if (key === validator.toString()) {
    validatorRewardPoints = Number(value);
    
    // No need to continue since we have found the validator.
    break;
  }
}
```

Regardless, if it's an object and we only apply the assignment for a specific key, can't we directly access the value `rewards.individual[validator.toString()]` or am I missing something?